src/pages/Chat/steps/ApartmentTypeInput.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  TranslationsConsumer,
+  TranslationsPlaceholderConsumer,
+} from '@hedviginsurance/textkeyfy'
+import {
+  UserResponse,
+  UserSelectInput,
+  UserTextInput,
+} from 'components/userInput/UserResponse'
+import * as React from 'react'
+import * as yup from 'yup'
+import { ValidationErrorMaybe } from '../components/validationErrors'
+import {
+  ApartmentType,
+  ChatContainer,
+  Effects as ChatActions,
+  LivingSituationState,
+} from '../state'
+
+interface ApartmentTypeInputProps {
+  appear?: boolean
+  onSubmit?: () => void
+  isCurrentMessage?: boolean
+}
+
+const handleChange = <K extends keyof LivingSituationState>(
+  field: K,
+  chatState: ChatActions,
+  format: (val?: string | number) => undefined | string | number = (value) =>
+    value,
+) => (event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
+  chatState.setLivingSituationProp(field, format(event.target.value))
+}
+
+const validationSchema = yup
+  .object<Partial<LivingSituationState>>({
+    apartmentType: yup
+      .mixed()
+      .oneOf(Object.keys(ApartmentType))
+      .required(),
+    size: yup
+      .mixed()
+      .test({
```

Comment:
What's up here? 🙂

Reply:
This can be an empty string _or_ a number, and it may not be over 250. So it's quite tricky to validate since emptystring becomes NaN if we make it a number, so this is the workaround :)

Reply:
Isnt empty string an invalid case though? Given that we `.test()` away that.

Reply:
sorry, i'm mixing up possible states and valid states when trying to explain it 😅

the input _could be_:
a) an empty string
b) a number

the input _is valid when_ it's:
a) a number
b) > 0
c) < 250

Reply:
the alternative would be to make the input _number only_, but that would mean we need to start at 0 and they can't really remove the 0, which isn't pretty for the end user

Reply:
(after trying to get rid of this, following an irl discussion)
We cant get rid of this, because `empty string` -> `yup.number()` = ``size must be a `number` type, but the final value was: `NaN` (cast from the value `""`).`` :(

Reply:
That's all I need to hear - lets keep it as is! :+1: